### PR TITLE
Battle-parity suite hygiene: mirror cooldownRecovery fact, pin replay-to-offset equivalence, fix stale comments

### LINKED ISSUE #2063: Battle-parity suite hygiene: mirror the FE-only cooldown fact, pin replayToOffset equivalence, fix stale draw-order comments
## Context

Found in the July 2026 health review (battle-parity pass). The 68-row scenario matrices are row-for-row aligned and value-identical down to FP artifacts — three hygiene gaps remain, batchable in one PR:

1. **FE-only supplementary fact:** the FE `cooldownRecovery` derived-stats fact (`UI/src/tests/lib/battle/battle-simulation-parity.test.ts:1689` — MaxHealth 900 / Toughness 60 / cdMultiplier 1.26) has no backend twin, while the other 4 facts are mirrored exactly (`Game.Core.Tests/Battle/BattleSimulatorParityTests.cs:1339-1410`). A backend change to the cooldown-channel derived stats would fail only the FE fact (or the FE fact gets deleted as "extra"), weakening the row-for-row audit trail CLAUDE.md mandates. Add the matching `[Fact]`.

2. **No mirrored pin for `replayToOffset` equivalence:** frontend.md asserts a battle fast-forwarded to `elapsedOffsetMs` (`battle-engine.ts:229-244`) replays identically to the backend's own tick loop (#1595/#1596/#1597), but no shared test row pins "run to offset X headless, then to conclusion == uninterrupted run / backend result" (FE-only coverage exists in `battle-engine.test.ts:440-606`; `BattleSimulatorTests.cs` has no offset scenario). A future refactor of either loop's boundary condition (`totalMs &lt;= cappedOffsetMs`, or the `totalMs - tickSize` exit) could shift the resume point one tick without any parity test failing. Add one mirrored row per side: simulate N ticks, snapshot, continue; assert equal to a straight run (FE) and `Simulate(maxMs = offset)` + continuation equals a full `Simulate` (BE).

3. **Stale draw-order comments (spec risk):** `BattleSimulatorParityTests.cs:578-579` and `battle-simulation-parity.test.ts:801-804` (`drawOrderMultiSkill`) still describe "two dodge draws" — the rule since #1457 is **3 draws per enemy fire**. Also `BattleSimulatorParityTests.cs:914` says "Str climbing 10→50" where the FE's "10→60" (`:1161`) is correct. These comments are the human-readable spec of the RNG parity contract; a developer extending the draw order from the stale comment would produce a divergent stream the outcome-insensitive scenario wouldn't catch.

### PR DESCRIPTION
**Pull request by ginderjeremiah**
## Summary

Three battle-parity test-suite hygiene fixes from the July 2026 health review (#2063), batched in one PR since each is small and independent:

1. **Mirrored the FE-only `cooldownRecovery` derived-stats fact.** The frontend suite pinned `cooldownRecovery`'s MaxHealth/Toughness/CDR/CooldownBonus/CooldownBonusMultiplier/`cdMultiplier` values, but the backend had no matching `[Fact]` — added `Parity_CooldownRecovery_ResolvesDerivedStats` mirroring it exactly, closing the gap in the row-for-row audit trail CLAUDE.md mandates for battle logic.

2. **Added a mirrored `replayToOffset` boundary pin on both sides.** Neither suite had a test asserting that simulating to a mid-battle offset and then continuing on the same battlers reaches the identical outcome/`totalMs` as one uninterrupted run. Added `Parity_RunToOffsetThenContinue_MatchesUninterruptedRun` (backend) and its frontend twin, both built on the existing deterministic `cooldownRecovery` scenario (0% crit/dodge/parry, so the RNG stream restart on the continuation call doesn't affect the outcome) — splitting `Simulate(offset)` then `Simulate()` again on the same simulator instance and comparing against a fresh, uninterrupted `Simulate()`. This guards the tick loop's boundary arithmetic (`totalMs <= limit`, the `totalMs - tickSize` exit) against a future refactor silently shifting the resume point by a tick.

3. **Fixed two stale comments** (present identically in both suites):
   - `drawOrderMultiSkill`: said "two dodge draws, one each now that Block's second draw is gone" — the rule since #1457 is **three** unconditional draws per enemy fire (parry, dodge, counter-crit), so two enemy skills firing is six draws, not two.
   - `grantedSkillWithEffects`: backend said "Str climbing 10→50" where the frontend's "10→60" (reflecting Strength after all five +10 stacks apply) is correct.

## Test plan

- [x] `dotnet build` — succeeds, 0 warnings/errors
- [x] `dotnet test Game.Core.Tests` — 1376 passed, 0 failed (includes the two new Facts)
- [x] `npx vitest run src/tests/lib/battle/battle-simulation-parity.test.ts` — 74 passed (includes the new test)
- [x] `npm run test:unit:ci` (full UI suite) — 3791 passed, 0 failed
- [x] `npm run lint` — clean (ESLint + svelte-check, 0 errors/warnings)

Closes #2063

---
_Generated by [Claude Code](https://claude.ai/code/session_01QW8uEw4Nem6t6TMwAKbT5K)_

### CODE REVIEW

File: Game.Core.Tests/Battle/BattleSimulatorParityTests.cs [hunk 1]
```diff
@@ -1357,6 +1357,63 @@ public void Parity_DoubleDerivedStats_ProducesShorterBattle()
...
+            var splitSim = new BattleSimulator(
+                Scenarios["cooldownRecovery"].Player(), Scenarios["cooldownRecovery"].Enemy(), ParitySeed);
+            var partial = splitSim.Simulate(offset);
+            Assert.False(partial.Victory);
+            Assert.False(partial.PlayerDied);
+            Assert.Equal(offset, partial.TotalMs);
+
+            // Continuing on the SAME (already-mutated) battlers, unbounded, must reach the identical
+            // conclusion a straight, uninterrupted run reaches from a fresh pair.
+            var continuation = splitSim.Simulate();
```
Comment (ginderjeremiah) [inline Game.Core.Tests/Battle/BattleSimulatorParityTests.cs:1412]: Nit: the correctness of this whole test rests on a load-bearing property that isn't stated here — `Simulate()` re-seeds a fresh `Mulberry32` on every call (`BattleSimulator.cs:22`), so the continuation does **not** resume the RNG stream; it restarts it. That only reaches the same conclusion because `cooldownRecovery` has no CriticalChance/DodgeChance/ParryChance on either battler, so every parry/dodge/crit draw fails regardless of the value drawn. If someone later gives this scenario a crit/dodge chance (or repoints the test at a different scenario), the re-seed would diverge and the test would fail confusingly — for a reason unrelated to the boundary arithmetic it claims to guard. Worth a one-line note that the scenario must be RNG-outcome-insensitive for the re-seeded continuation to be valid. Same applies to the FE twin.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: UI/src/tests/lib/battle/battle-simulation-parity.test.ts [hunk 2]
```diff
@@ -1703,6 +1703,35 @@ describe('Battle simulation parity with backend', () => {
...
+
+		const splitSim = new BattleSimulator(scenario.player(), scenario.enemy(), PARITY_SEED);
+		const partial = splitSim.simulate(offset);
+		expect(partial.victory).toBe(false);
+		expect(partial.playerDied).toBe(false);
+		expect(partial.totalMs).toBe(offset);
+
+		// Continuing on the SAME (already-mutated) battlers, unbounded, must reach the identical conclusion a
+		// straight, uninterrupted run reaches from a fresh pair.
+		const continuation = splitSim.simulate();
```
Comment (ginderjeremiah) [inline UI/src/tests/lib/battle/battle-simulation-parity.test.ts:1731]: Nit (mirror of the backend comment): `simulate()` re-seeds a fresh `Mulberry32` each call (`battle-simulator.ts:34`), so this continuation restarts the RNG stream rather than resuming it. The re-run only lands on the same outcome because `cooldownRecovery` sets no CriticalChance/DodgeChance/ParryChance, so every draw fails regardless of value. A one-line note that the chosen scenario must be RNG-outcome-insensitive would stop a future edit (adding crit to this scenario, or picking a different one) from breaking the test for a reason unrelated to the boundary arithmetic it guards.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Review by ginderjeremiah — COMMENTED** [review]
## Review: Battle-parity suite hygiene (mirror cooldownRecovery fact, pin replay-to-offset equivalence, fix stale comments) — closes #2063

**Verdict: Approve** — all three hygiene items from #2063 are implemented correctly and verified against the source. One minor nit only (below); not a blocker. _(Submitted as a comment because I'm the PR author and GitHub won't accept a gating self-review; the real call is Approve.)_

I verified each claim rather than taking the description at face value:

- **cooldownRecovery derived-stats fact** — the new backend `Parity_CooldownRecovery_ResolvesDerivedStats` mirrors the FE fact (`battle-simulation-parity.test.ts:1689`) value-for-value: MaxHealth 900 / Toughness 60 / CooldownRecovery 1.0 / CooldownBonus 0.25 / CooldownBonusMultiplier 1.04 / cdMultiplier 1.26, enemy 400/30. Closes the row-for-row audit gap CLAUDE.md mandates.
- **replay-to-offset pin** — checked both `Simulate` loops: each re-seeds one `Mulberry32` and mutates the battlers in place, and returns per-call `totalMs`, so `Simulate(2400)` + continuation == a straight `Simulate()` is sound arithmetic. offset 2400 is tick-aligned (MsPerTick 40 → 60 ticks) and genuinely mid-battle (cooldownRecovery concludes at 5760ms, player victory), so the `partial` assertions (`!victory`, `!playerDied`, `totalMs == 2400`) are correct. FE twin matches exactly.
- **stale comments** — the corrected draw-order comment ("three draws each — parry, dodge, counter-crit, per #1457 — six total") matches the actual rule in `battle-step.ts:180`/`BattleContext.cs:226`; the old "two dodge draws" text was genuinely stale. The Str `10→60` fix aligns the backend `grantedSkillWithEffects` comment with the FE.

### Nits

- The replay-to-offset test's validity rests on an unstated, load-bearing property: `Simulate()` **re-seeds** the RNG each call, so the continuation restarts the stream rather than resuming it — it only reaches the same conclusion because `cooldownRecovery` has no CriticalChance/DodgeChance/ParryChance, making every draw outcome-insensitive. A one-line note to that effect (both sides) would keep a future edit from breaking the test for a reason unrelated to the boundary arithmetic it guards. Inline on `BattleSimulatorParityTests.cs:1408` and `battle-simulation-parity.test.ts:1727`.

### Test coverage

Exemplary for this project's rules: the change is entirely test hygiene, and both new tests are mirrored across the FE and BE suites as CLAUDE.md requires for battle logic. The two `[Fact]`/`it` pairs assert identical scenarios and expected values on each side.

### What's good

Genuinely careful work — each new pin is built on the deterministic `cooldownRecovery` scenario for a reason, the comments explain the boundary arithmetic and the #-referenced rules they encode, and the stale-comment fixes were the right catch (these comments are the human-readable spec of the RNG parity contract). Scope is tight and well-batched.

I'll hold off on merging: CI checks haven't reported yet (status pending), and I'd like the nit note considered first — your call whether to add it here or wave it through.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Comment by ginderjeremiah** [comment]
## Re-review — nit resolved, holding for CI

Re-reviewed the delta since the prior Approve (`b42348e9` → `612fc212`). The only change is the comment-only commit that adds the RNG re-seed note the sole nit asked for, on both sides:

> `Simulate()`/`simulate()` re-seed a fresh RNG each call rather than resuming the prior stream, so the continuation only matches an uninterrupted run because `cooldownRecovery` has no crit/dodge/parry chance to make a draw's value matter.

That's accurate and load-bearing exactly where the nit flagged, so the one open item is closed. **Verdict: Approve, no outstanding nits or should-fixes**, no follow-up issues needed — this is tight, well-batched test hygiene.

CI: run #2629 on `b42348e9` passed; run #2630 on the current head (identical code, comment-only) is still in progress. Once it goes green I'll merge.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Comment by ginderjeremiah** [comment]
CI run #2630 on the current head came back green ✅ — **merged** (squash → `39e47ff5`). Closes #2063. Thanks for the tidy hygiene work.

---
_Generated by [Claude Code](https://claude.ai/code)_
